drivers/sensor/icm42688/icm42688_emul.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT invensense_icm42688
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/emul.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/drivers/spi_emul.h>
+#include <zephyr/logging/log.h>
+
+#include <icm42688_reg.h>
+
+LOG_MODULE_DECLARE(ICM42688, CONFIG_SENSOR_LOG_LEVEL);
+
+#define NUM_REGS (UINT8_MAX >> 1)
```

Comment:
Sort of :-)

The part has register banks (tdk/invensense seem to do this a lot) which can be selected by writing to an always available register. This will likely need to be updated when/if mocking that sort of pattern is ever needed. Clearly works fine for now though, not a blocker.

Reply:
Thanks, I'm working on the interrupt support for the driver and emulator now so I'll try to squeeze this in there.